### PR TITLE
fix: replace component values with actual onces before rendering []

### PR DESCRIPTION
**Pull request by Chaoste**
As shown here in [this Miro board](https://miro.com/app/board/uXjVNL-mFho=/?moveToWidget=3458764570954909785&cot=14), we want to replace the component value pointers with the actual unbound value or bound value. We now replace those correctly, so in the final rendering we don't have to handle `ComponentValue` anymore.

### CODE REVIEW

**Comment by dimitrycf** [comment]
GH doesn't allow me to comment on non-changed LOC, so pasting this comment here

image

**Comment by dimitrycf** [comment]
And this 
image

**Comment by dimitrycf** [comment]
And this one

image

**Comment by Chaoste** [comment]
@dimitrycf all those places should be fine as we won't face a design component in those cases but always a developer component. In `resolveDesignComponent` at the top of this component, we replace it not only the node with the actual ones but also replace the component values on the nodes with the actual unbound or bound values. Thus, the following don't has to deal with "ComponentValues".

If a design component resolves to a button deeper in the tree and the button is missing a value, it will take the defaultValue from the component definition of the button, not the design component.

**Comment by SofiaMargariti** [comment]
Not sure if this is related to this PR but in preview/delivery mode it doesn't seem to render any values for design components:
Editor:
Screenshot 2023-12-14 at 10 49 24
Preview/Delivery:
Screenshot 2023-12-14 at 10 49 48

**Review by dimitrycf — APPROVED** [review]
Thanks Thomas for addressing my questions! 🙏

**Comment by SofiaMargariti** [comment]
I am also getting this error when dropping the same component a second time:
Screenshot 2023-12-14 at 10 56 16
Checking if it also happening on main..

**Comment by dimitrycf** [comment]
> I am also getting this error when dropping the same component a second time: Screenshot 2023-12-14 at 10 56 16 ~Checking if it also happening on main..~ I don't see it happening on main so probably related to these changes.

I tried reproducing this by dropping same component twice but it worked for me 🤷 
https://www.loom.com/share/aac4f5451d7146d18e279f62199ec76c

File: packages/experience-builder-sdk/src/blocks/editor/VisualEditorBlock.tsx [hunk 1]
```diff
@@ -149,18 +149,6 @@ export const VisualEditorBlock = ({
             ...acc,
             [variableName]: transformContentValue(value, variableDefinition),
           };
```
[the remark this answers is not in the document]
Comment (Chaoste) [reply packages/experience-builder-sdk/src/blocks/editor/VisualEditorBlock.tsx:151]: There is a misunderstanding. We replace the ComponentValues in the deserialisation process. Here, you will only see DesignVariable/ BoundValue/ UnboundValue.
Keep in mind that the ComponentValue can now also be replaced by a BoundValue.

**Comment by Chaoste** [comment]
> I am also getting this error when dropping the same component a second time:

@SofiaMargariti this sounds related to the Bug fixes in user interface that I just merged. If you pull the laster master for user interface, do you still get the error?

**Comment by Chaoste** [comment]
> Not sure if this is related to this PR but in preview/delivery mode it doesn't seem to render any values for design components

@SofiaMargariti I tried to only address the editor logic in this PR. I'd like to file another one for preview logic as follow up but I don't foresee it as blocker for Telus to play around with it :)

**Review by SofiaMargariti — APPROVED** [review]

**Pull request merged**